### DISCUSSION TITLE
Offline maps and high polyline count

### DISCUSSION THREAD

**Discussion by jeanplevesque** [Mapsui/Mapsui#2360] [Q&A]
Hello, I'm considering Mapsui for a mobile project (iOS, Android) and I'd like to confirm the feasibility of a few things.
I already checked the documentation and past issues, but some points are still unclear to me.

1. Offline maps using non-bitmap tiles.
   - I'd like to download map data (not sure yet of the provider, perhaps MapBox) in a format that doesn't take too much space (for bandwidth and disk space considerations).
1. Draw thousands of polylines.
   - I'd like to be able to draw a high count of polylines (to represent trails) and keep good performance (FPS) when manipulating the map (zooming, panning).
   - I already have algorithms to adjust the level of details based on zoom.
     - When zoomed out, I can go as low as 700 polylines with a total of 3000 points.
     - When zoomed in, my data has 2300 polylines with a total of 1 million points.

Thanks! 😃

**Comment by pauldendulk** [comment]
We don't have something specific for storing vector data on mobile. Yet, we use [NTS](https://github.com/NetTopologySuite/NetTopologySuite) for vector data. So you could use something storing NTS to an sqlite database. For instance [this library](https://www.nuget.org/packages/Microsoft.EntityFrameworkCore.Sqlite.NetTopologySuite/) maintained by Microsoft. You could also ask this question in the NTS project.

You will need to implement your own IProvider in which you use this logic. This should not be to hard. It is two methods and a property: https://mapsui.com/api/Mapsui.Providers.IProvider.html

About drawing many polylines. You could look at the performance samples. Select 'Performance' in the top dropdown here: https://mapsui.com/samples/

You should probably use the 'RasterizingTileLayer with LineString'.

![image](https://github.com/Mapsui/Mapsui/assets/963462/8a4ca054-1491-47b9-b885-3ac4f517a184)

**Reply by charlenni** [reply]
I have an addon for Mapsui to display vector maps, but it is missing street names. The rest seems to work. It could display Mapbox formatted maps.

**Reply by jeanplevesque** [reply]
Thanks for the replies! I'll check this out!
